### PR TITLE
[FLINK-13505][docs-zh] Translate "Java Lambda Expressions" page into Chinese

### PR DESCRIPTION
**Pull request by YngwieWang**
Capitalized all the "lambda".
PTAL @klion26 @wuchong

### CODE REVIEW

File: docs/dev/java_lambdas.zh.md [hunk 1]
```diff
@@ -22,41 +22,37 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-Java 8 introduced several new language features designed for faster and clearer coding. With the most important feature,
-the so-called "Lambda Expressions", it opened the door to functional programming. Lambda expressions allow for implementing and
-passing functions in a straightforward way without having to declare additional (anonymous) classes.
+Java 8 引入了几种新的语言特性，旨在实现更快、更清晰的编码。 作为最重要的特性，即所谓的“Lambda 表达式”，它开启了函数式编程的大门。Lambda 表达式允许以简捷的方式实现和传递函数，而无需声明额外的（匿名）类。
```
Comment (klion26) [inline docs/dev/java_lambdas.zh.md]: 即所谓的“Lambda 表达式” -> 即所谓的 “Lambda 表达式”?

File: docs/dev/java_lambdas.zh.md [hunk 2]
```diff
@@ -22,41 +22,37 @@ specific language governing permissions and limitations
...
 
-<span class="label label-danger">Attention</span> Flink supports the usage of lambda expressions for all operators of the Java API, however, whenever a lambda expression uses Java generics you need to declare type information *explicitly*. 
+<span class="label label-danger">注意</span> Flink 支持对 Java API 的所有算子使用 Lambda 表达式，但是，当 Lambda 表达式使用 Java 泛型时，你需要*显式*声明类型信息。
```
Comment (klion26) [inline docs/dev/java_lambdas.zh.md]: maybe "要*显式*声" need change to "要 *显式* 声", you can verify this locally according to the [instructions](https://github.com/apache/flink/blob/master/docs/README.md)

File: docs/dev/java_lambdas.zh.md [hunk 3]
```diff
@@ -22,41 +22,37 @@ specific language governing permissions and limitations
...
-Unfortunately, functions such as `flatMap()` with a signature `void flatMap(IN value, Collector<OUT> out)` are compiled into `void flatMap(IN value, Collector out)` by the Java compiler. This makes it impossible for Flink to infer the type information for the output type automatically.
+不幸的是，`flatMap()` 这样的函数，它的签名 `void flatMap(IN value, Collector<OUT> out)` 被 Java 编译器编译为 `void flatMap(IN value, Collector out)`。这样 Flink 就无法自动推断输出的类型信息了。
 
-Flink will most likely throw an exception similar to the following:
+Flink 很可能抛出类似如下的异常：
 
 {% highlight plain%}
 org.apache.flink.api.common.functions.InvalidTypesException: The generic type parameters of 'Collector' are missing.
-    In many cases lambda methods don't provide enough information for automatic type extraction when Java generics are involved.
+    In many cases Lambda methods don't provide enough information for automatic type extraction when Java generics are involved.
```
Comment (klion26) [inline docs/dev/java_lambdas.zh.md]: why do we need to change this one? if we change this one, then the English version should be updated also.

File: docs/dev/java_lambdas.zh.md [hunk 4]
```diff
@@ -22,41 +22,37 @@ specific language governing permissions and limitations
...
     An easy workaround is to use an (anonymous) class instead that implements the 'org.apache.flink.api.common.functions.FlatMapFunction' interface.
     Otherwise the type has to be specified explicitly using type information.
 {% endhighlight %}
 
-In this case, the type information needs to be *specified explicitly*, otherwise the output will be treated as type `Object` which leads to unefficient serialization.
+在这种情况下，需要*显式*指定类型信息，否则输出将被视为 `Object` 类型，这会导致低效的序列化。
```
Comment (klion26) [inline docs/dev/java_lambdas.zh.md]: maybe "要*显式*指" -> "要 *显式* 指"

File: docs/dev/java_lambdas.zh.md (hunk 1 above)
Comment (YngwieWang) [reply docs/dev/java_lambdas.zh.md]: 《Flink Translation Specifications》上面说：中文标点符号前后不需要跟空格，即使前后挨着的是英文单词。

File: docs/dev/java_lambdas.zh.md (hunk 2 above)
Comment (YngwieWang) [reply docs/dev/java_lambdas.zh.md]: 我看了那篇 instruction， 非常有帮助，不过没有写关于中文的斜体前后是否空格。我认为你的建议比较好，没有空格的话斜体字可能会和后面的字重叠，我修改了，谢谢!

File: docs/dev/java_lambdas.zh.md (hunk 3 above)
Comment (YngwieWang) [reply docs/dev/java_lambdas.zh.md]: 我没太明白你的意思，这部分是异常信息，原文我没有翻译。可以再进一步说明一下么？谢谢。

**Comment by YngwieWang** [comment]
> @YngwieWang thanks for your contribution, I left some comments

Thanks for your review very much. I made some changes. PTAL

File: docs/dev/java_lambdas.zh.md (hunk 4 above)
Comment (YngwieWang) [reply docs/dev/java_lambdas.zh.md]: I fix it ,thanks.

File: docs/dev/java_lambdas.zh.md (hunk 3 above)
Comment (klion26) [reply docs/dev/java_lambdas.zh.md]: 这个地方为什么需要将 "lambda" 改成 "Lambda" 呢？如果需要的话，是不是英文版本也需要修改

Comment (YngwieWang) [reply docs/dev/java_lambdas.zh.md]: 非常感谢，你真细心。因为我翻译的时候统一把“lambda”翻译成“Lambda”，可能是不小心整体替换了，已经修正了。另外你觉得翻译的 lambda 全部首字母大写是否合适？因为我觉得翻译以后有的大写有的小写不统一的话有可能造成困惑。再次感谢！

File: docs/dev/java_lambdas.zh.md [hunk 5]
```diff
@@ -22,32 +22,28 @@ specific language governing permissions and limitations
...
-the so-called "Lambda Expressions", it opened the door to functional programming. Lambda expressions allow for implementing and
-passing functions in a straightforward way without having to declare additional (anonymous) classes.
+Java 8 引入了几种新的语言特性，旨在实现更快、更清晰的编码。 作为最重要的特性，即所谓的“Lambda 表达式”，它开启了函数式编程的大门。Lambda 表达式允许以简捷的方式实现和传递函数，而无需声明额外的（匿名）类。
 
-<span class="label label-danger">Attention</span> Flink supports the usage of lambda expressions for all operators of the Java API, however, whenever a lambda expression uses Java generics you need to declare type information *explicitly*. 
+<span class="label label-danger">注意</span> Flink 支持对 Java API 的所有算子使用 Lambda 表达式，但是，当 Lambda 表达式使用 Java 泛型时，你需要 *显式* 声明类型信息。
 
-This document shows how to use lambda expressions and describes current limitations. For a general introduction to the
-Flink API, please refer to the [Programming Guide]({{ site.baseurl }}/dev/api_concepts.html)
+本文档介绍了如何使用 Lambda 表达式并描述了其在当前应用中的限制。有关 Flink API 的一般性介绍， 请参阅[编程指南]({{ site.baseurl }}/zh/dev/api_concepts.html)。
```
Comment (wuchong) [inline docs/dev/java_lambdas.zh.md]: "一般性" -> "通用" ?

Comment (YngwieWang) [reply docs/dev/java_lambdas.zh.md]: This PR is closed. I don't know why. should I open a new one?

Comment (wuchong) [reply docs/dev/java_lambdas.zh.md]: No, you don't need. This PR is merged by me.

Comment (YngwieWang) [reply docs/dev/java_lambdas.zh.md]: OK, thanks very much. It's my pleasure to make contributes.

**Pull request closed without merging**